h2o-core/src/main/java/water/rapids/ast/prims/search/AstWhichFunc.java

```diff
@@ -0,0 +1,225 @@
+                    } else {
+                        if (row[i] < val) {
+                            val = row[i];
+                            valIndex = i;
+                        }
+                    }
+                }
+            }
+            else{
+                throw new IllegalArgumentException("Incorrect argument: expected to search for max() or min(), received " + searchVal());
+            }
+            return new ValRow(new double[]{valIndex}, null);
+        } else
+            throw new IllegalArgumentException("Incorrect argument: expected a frame or a row, received " + val1.getClass());
+    }
+
+
+    /**
+     * Compute row-wise, and return a frame consisting of a single Vec of value indexes in each row.
+     */
+    private ValFrame rowwiseWhichVal(Frame fr, final boolean na_rm) {
+        String[] newnames = {"which." + searchVal()};
+        Key<Frame> newkey = Key.make();
+
+        // Determine how many columns of different types we have
+        int n_numeric = 0, n_time = 0;
+        for (Vec vec : fr.vecs()) {
+            if (vec.isNumeric()) n_numeric++;
+            if (vec.isTime()) n_time++;
+        }
+        // Compute the type of the resulting column: if all columns are TIME then the result is also time; otherwise
+        // if at least one column is numeric then the result is also numeric.
+        byte resType = n_numeric > 0 ? Vec.T_NUM : Vec.T_TIME;
+
+        // Construct the frame over which the val index should be computed
+        Frame compFrame = new Frame();
+        for (int i = 0; i < fr.numCols(); i++) {
+            Vec vec = fr.vec(i);
+            if (n_numeric > 0? vec.isNumeric() : vec.isTime())
+                compFrame.add(fr.name(i), vec);
+        }
+        Vec anyvec = compFrame.anyVec();
+
+        // Take into account certain corner cases
+        if (anyvec == null) {
+            Frame res = new Frame(newkey);
+            anyvec = fr.anyVec();
+            if (anyvec != null) {
+                // All columns in the original frame are non-numeric -> return a vec of NAs
+                res.add("which." + searchVal(), anyvec.makeCon(Double.NaN));
+            } // else the original frame is empty, in which case we return an empty frame too
+            return new ValFrame(res);
+        }
+        if (!na_rm && n_numeric < fr.numCols() && n_time < fr.numCols()) {
+            // If some of the columns are non-numeric and na_rm==false, then the result is a vec of NAs
+            Frame res = new Frame(newkey, newnames, new Vec[]{anyvec.makeCon(Double.NaN)});
+            return new ValFrame(res);
+        }
+
+        // Compute over all rows
+        final int numCols = compFrame.numCols();
+        Frame res = new MRTask() {
+            @Override
+            public void map(Chunk[] cs, NewChunk nc) {
+                for (int i = 0; i < cs[0]._len; i++) {
+                    int numNaColumns = 0;
+                    double value = Double.NEGATIVE_INFINITY;
```

Comment:
Are you sure this (the value variable) should be a single value?  A Chunk has multiple rows